pkg/apis/autoscaling/v1alpha1/pa_lifecycle.go

```diff
@@ -76,10 +76,16 @@ func (pa *PodAutoscaler) annotationFloat64(key string) (float64, bool) {
 
 // ScaleBounds returns scale bounds annotations values as a tuple:
 // `(min, max int32)`. The value of 0 for any of min or max means the bound is
-// not set
+// not set.
+// Note: min will be ignored if the PA is not reachable
+// TODO(tanzeeb) Tests!
 func (pa *PodAutoscaler) ScaleBounds() (min, max int32) {
-	return pa.annotationInt32(autoscaling.MinScaleAnnotationKey),
-		pa.annotationInt32(autoscaling.MaxScaleAnnotationKey)
+	if pa.Spec.Reachable {
+		min = pa.annotationInt32(autoscaling.MinScaleAnnotationKey)
+	}
```

Comment:
I am curious what this means for the initial readiness of the KPA?  Upon creation the KPA is never reachable, so does `minScale: 5` still mean that the Revision doesn't report initial readiness until 5 pods come up?

Reply:
I concur. We should have a look through all calling sites of this and determine if we really want all of them to observe this.

Reply:
@mattmoor: Since the KPA won't be reachable, the minimum will be 1, so the KPA will be `active` once 1 pod comes up. Once the KPA becomes reachable, the minimum will be 5, it will be marked `activating` until 5 pods come up.

Reply:
@markusthoemmes: In previous iterations of this PR, I did a search and found that all non-test usage of `ScaleBounds` are confined to the KPA and HPA, and every case that looked at `ScaleBounds` also needed to take `Reachable` into account. There was a case that used the `minScale` annotation directly, but it also needed to take `Reachable` into account and should actually have used `ScaleBounds` anyway. Tagged you on the PR #5105.

Reply:
I think this is a place where we could abuse a tri-state `Reachable` field (`*bool`).

1. Until initial readiness has been established, leave `Reachable: nil`
2. After initial readiness has been established, do what you do.

With my other comment about conservative behavior in the presence of unknown reachability, this should get us both semantics we want here.  🤞

Reply:
@mattmoor Just to clarify, I don't think we need to change anything here.

>  so does minScale: 5 still mean that the Revision doesn't report initial readiness until 5 pods come up?

No, the revision will report ready when 1 pod comes up because of [this code](https://github.com/knative/serving/pull/4884/files#diff-d155aed549091184959c9607a39bc1e7R277-R283) in the KPA reconciler. The scenario is tested in the MinScale readiness tests [here](https://github.com/knative/serving/pull/4884/files#diff-fb83cfcb029a8fcb90576e370d749f23R68-R73).

The tri-state is still probably usable for the unknown reachibility scenario.

Reply:
Wouldn't that code return minScale when reachability is unknown?

Reply:
> Wouldn't that code return minScale when reachability is unknown?

Yes, but reachability can only be `false` here, right?

Reply:
Oh, i think I see what you're saying... if a revision is in the middle of activating, and we upgrade serving during that window before it has activated, it will have to scale to `minScale` instead of 1.

Reply:
Correcting a misunderstanding of mine: It is desirable for us to scale to 5 during activation.

Reply:
it is actually pretty cool that this ends up being the sole functional change in this PR.  Thanks for breaking this down so much.